### ISSUE TITLE
Result varied w/o include-ext while git diff

### ISSUE THREAD

**Issue by seognil** [AlDanial/cloc#385]
Thanks for the great tool!

But I saw some confusing result when I use `cloc --git --diff` with or without the `--include-ext` option

I think maybe `--include-ext` missed some files when it comes `deleted or added files` between two commits

`json` of `package.json` is missing, and `js` count is less than the full run.

(I didn't dive into the cloc source code yet)
(and also I noticed that hidden dotfile like `./.babelrc` is not analyzed (is that optional?))

environment: (Does arguments order matter at MacOS?)
```
OS: 64bit Mac OS X 10.14.4 18E226
Kernel: x86_64 Darwin 18.5.0
zsh 5.3 (x86_64-apple-darwin18.0)
```

```
$  cloc --git --diff d92413965 e3c1346d3 --include-ext=babelrc,js,json,md
    7275 text files.
    7268 text files.s
   14785 files ignored.

github.com/AlDanial/cloc v 1.80  T=21.02 s (0.0 files/s, 0.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
JavaScript
 same                            0              0              0              0
 modified                        0              0              0              0
 added                          11             91             65            446
 removed                        18             68            166           1707
Markdown
 same                            0              0              0              0
 modified                        0              0              0              0
 added                           4             51              0            327
 removed                         1              4              0              4
-------------------------------------------------------------------------------
SUM:
 same                            0              0              0              0
 modified                        0              0              0              0
 added                          15            142             65            773
 removed                        19             72            166           1711
-------------------------------------------------------------------------------
```

```
$  cloc --git --diff d92413965 e3c1346d3
    7275 text files.
    7268 text files.s
    2733 files ignored.

github.com/AlDanial/cloc v 1.80  T=90.35 s (0.0 files/s, 0.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
JavaScript
 same                         4074         179274         294079        1337683
 modified                       12              0             90            327
 added                          11            165             72            494
 removed                        18             76            343           2213
JSON
 same                          212             14              0          43472
 modified                        1              0              0              7
 added                           0              0              0             22
 removed                         0              0              0              9
Markdown
 same                           38            381              0           1007
 modified                        2              0              0              0
 added                           4             58              0            334
 removed                         1              4              0              4
... (other types with no difference)
-------------------------------------------------------------------------------
SUM:
 same                         5948         240202         310607        1954205
 modified                       15              0             90            334
 added                          15            223             72            850
 removed                        19             80            343           2226
-------------------------------------------------------------------------------
```

---

UPDATE:
I figure out that the `include-lang` option is good as well,
And I check the cloc source code a bit, maybe some missing file types cause the problem,
such as `es` points to `JavaScript` too.
(Also tried on ubuntu by apt install, `include-lang` is good, but no `include-ext` option there)

(But I do another check of the types, no other types, weird...)
```
$   git diff d92413965 e3c1346d3 --name-status | sed -E "s/.*\.(.*)$/\1/" | sort | uniq |  paste -sd "," -
# => babelrc,js,json,md
```

```
$ cloc --git --diff d92413965 e3c1346d3 --include-lang=JavaScript,Markdown,JSON
    7275 text files.
    7268 text files.s
    6107 files ignored.

github.com/AlDanial/cloc v 1.80  T=77.51 s (0.0 files/s, 0.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
JavaScript
 same                         4074         179274         294079        1337683
 modified                       12              0             90            327
 added                          11            165             72            494
 removed                        18             76            343           2213
JSON
 same                          212             14              0          43472
 modified                        1              0              0              7
 added                           0              0              0             22
 removed                         0              0              0              9
Markdown
 same                           38            381              0           1007
 modified                        2              0              0              0
 added                           4             58              0            334
 removed                         1              4              0              4
-------------------------------------------------------------------------------
SUM:
 same                         4324         179669         294079        1382162
 modified                       15              0             90            334
 added                          15            223             72            850
 removed                        19             80            343           2226
-------------------------------------------------------------------------------
```

**Comment by AlDanial** [comment]
I take it the git repo you're working with isn't public?  I ran a few tests using the cloc repo but wasn't able to reproduce omitted files as you're seeing with JSON.  Here's my test:
<pre>
> cloc --git --diff --include-ext wast,c 1d5b3db284 acdf9f51d                                                              
github.com/AlDanial/cloc v 1.81  T=0.06 s (16.6 files/s, 16.6 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
WebAssembly
 same                            0              0              0              0
 modified                        0              0              0              0
 added                           0              0              0              0
 removed                         1              8             20             32
C
 same                            0              0              0              0
 modified                        0              0              0              0
 added                           0              0              0              0
 removed                         1              3             11              6
-------------------------------------------------------------------------------
SUM:
 same                            0              0              0              0
 modified                        0              0              0              0
 added                           0              0              0              0
 removed                         2             11             31             38
-------------------------------------------------------------------------------
</pre>
<pre>
> cloc --git --diff 1d5b3db284 acdf9f51d                                                                                   
github.com/AlDanial/cloc v 1.81  T=4.32 s (0.2 files/s, 0.2 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Perl
 same                            0              0           2502          22796
 modified                        4              0              0              0
 added                           0              0              0              0
 removed                         0              1              2             25
YAML
 same                            0              0              0              0
 modified                        0              0              0              0
 added                           0              0              0              0
 removed                         2              0              2             40
WebAssembly
 same                            0              0              0              0
 modified                        0              0              0              0
 added                           0              0              0              0
 removed                         1              8             20             32
C
 same                            0              0              0              0
 modified                        0              0              0              0
 added                           0              0              0              0
 removed                         1              3             11              6
-------------------------------------------------------------------------------
SUM:
 same                            0              0           2502          22796
 modified                        4              0              0              0
 added                           0              0              0              0
 removed                         4             12             35            103
-------------------------------------------------------------------------------
</pre>

It would be most helpful if you can demonstrate the problem on a code base I have access to.  For now I'll poke about the code a bit and see where ``--include-ext`` and ``--include-lang`` differ; the operations for both of those should be close to each other.

Regarding ``babelrc``, that is not a recognized extension.  To see ones you can use, do
``cloc --show-ext``.

**Comment by AlDanial** [comment]
Also: regarding argument order:  the order of switches does not matter, but the order of the two diff arguments does.  The one on the left, your ``d92413965``, is considered the old, or "before" code and the one on the right, ``e3c1346d3`` is the new or "after" code.

**Comment by seognil** [comment]
Oh sorry... its my personal project...
I would try to make a working demo code.

**Comment by seognil** [comment]
I made a script to reproduce the problem...

```
# clone the lodash source code
git clone https://github.com/lodash/lodash.git ; cd lodash

# choosing two commit by:
# git log --pretty=format:"%h - %cd"
# Let's see these two: 6b1d83c96 e8c87cfc4

# Check the types
git diff 6b1d83c96 e8c87cfc4 --name-status | grep -E '\.' | sed -E "s/.*\.(.*)$/\1/" | sort | uniq |  paste -sd "," -
# Resulting: js,json,md,yml

# And these two results is not equivalent
cloc --git --diff 6b1d83c96 e8c87cfc4 --include-ext=js,json,md,yml
cloc --git --diff 6b1d83c96 e8c87cfc4
```

---

And why I would use cloc this way?
Because I'm trending to analyze the only different files between two commit,
(And already get the detailed diff file list by `git diff --name-status`)

(I'm thinking an option for output, such as: --diff-minimize, to filter out the same types...)

**Comment by AlDanial** [comment]
Thanks, I can duplicate the problem now.  Will be a few days before a fix though.

**Comment by AlDanial** [comment]
Please give the cloc version committed with 66d2e51 a try, see if it reports what you expect.

**Comment by seognil** [comment]
Awesome, it works

I notice that my mac cloc version before this fix is 1.80 and the `66d2e51` bumped to 1.81.

The ubuntu apt version with no `exclude-ext` option I mentioned before is 1.74,
And I git clone the cloc repo (which is 1.81) on ubuntu manually, it works fine as well.

**Comment by AlDanial** [comment]
The odd numbered versions are for "in development".  I plan to release 1.82 in a couple of weeks, will include this fix.
